docs/design/settings/extensions/settings-extension-example.md

```diff
@@ -0,0 +1,156 @@
+
+```rust
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Clone)]
+struct MyDaemonSettingV1 {
+    query_url: Url,
+}
+
+impl SettingsModel for MyDaemonSettingV1 {
+    fn get_version() -> &'static str { /* */ }
+
+    fn set(current_value: Option<Self>, new_value: Self) -> Result<Self> { /* */ }
+    
+    // etc
+}
+```
+
+You can then refer to this model when invoking the `SettingsExtension` entrypoint to the SDK:
+
+```rust
+fn main() -> Result<()> {
+    bottlerocket_settings_sdk::SettingsExtension::with_models(vec![
+        BottlerocketSetting::<MyDaemonSettingsV1>::model(),
+    ])
+    .run_extension()
+    .context("Settings extension encountered an error.")
+}
+```
+
+
+## Configuration File Template
+In your variant repository, you will need to create an RPM package which installs `mydaemon` into your custom Bottlerocket variant.
+In this package, you will also want to include a templated configuration file, which is what the settings system interacts with:
+
+```
+├── packages
+│   └── mydaemon
+│       ├── build.rs
+│       ├── Cargo.toml
+│       ├── mydaemon.spec
+│       ├── mydaemon.template.yaml
+│       └── pkg.rs
+├── sources
+└── variants
+
+# mydaemon.template.toml
+---
+[required-extensions]
+mydaemon: v1
+---
+query-url: {{ mydaemon.query-url }}
+```
+The configuration template includes "frontmatter" which informs the settings system to use settings values owned by the `mydaemon` extension, then the rest of the template is used to render the configuration file.
+
+## Service File
+Changing settings typically results in changes to configuration file templates, which results in systemd services being restarted to trigger the change in behavior.
+You should specify a `mydaemon.service` systemd service file which will be installed into the Bottlerocket system and used to manage the lifecycle of your daemon.
+
+```
+├── packages
+│   └── mydaemon
+│       └── ...
+│       └── mydaemon.service
+├── sources
+└── variants
+```
+
+## Resulting Disk Layout
+We've discussed constructing a setting, using a configuration file template, and associating that with a service.
+The Bottlerocket build system must arrange these files on disk to produce an image file.
+This section discusses how the various artifacts will be arranged on a running system.
+
+### Settings Extension Binaries
+Extension binaries are stored under `sys-root/usr/libexec` and symlinked into `sys-root/usr/libexec/settings`.
+`sys-root/usr/share/settings` will also hold the config file for the settings extensions, which will be named `${SETTING_NAME}.toml`
+The setting owned by the extension will share a name with the symlink.
+
+
+For example, the extension for `my-setting` may be installed like so:
+```
+sys-root
+└── usr
+    ├── libexec
+    │   ├── my-setting-binary
+    │   └── settings
+    │       └── my-setting -> ../my-setting-binary
+    └── share
+        └── settings
+            └── my-setting.toml
```

Comment:
nit: It'd help to stick with the `mydaemon` example to show how it all fits together.

```suggestion
sys-root
└── usr
    ├── libexec
    │   ├── mydaemon-settings
    │   └── settings
    │       └── mydaemon-> ../mydaemon-settings
    └── share
        └── settings
            └── mydaemon.toml
```